### PR TITLE
TA #1 - Team Impasta: Using the Open Trivia Database API (Dayeon Kang, Shaunak Badani)

### PR DESCRIPTION
**Pull request by shaunak-badani**
- Script in python to fetch data from the Open Trivia Database API
- Using argparse, parameters can be passed to the command line to determine:
    - Which category of questions to fetch
    - How many questions to fetch.
- The script saves the questions in a csv file called "trivia_questions.csv"
- The columns of the dataset:
    - [type,difficulty,category,question,correct_answer,incorrect_answer_1,incorrect_answer_2,incorrect_answer_3]
-
Setting up the virtual environment: 
```
python3 -m venv ../ta1
source ../ta1/bin/activate
pip install -r requirements.txt
```

Running the program:
```
usage: python3 api_trivia.py [-h] [--category {9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32}] [-n N]
Get questions from the Open Trivia API
options:
  -h, --help            show this help message and exit
  --category {9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32}
                        {"9": "General Knowledge", "10": "Entertainment: Books", "11": "Entertainment: Film", "12": "Entertainment: Music", "13": "Entertainment: Musicals & Theatres",
                        "14": "Entertainment: Television", "15": "Entertainment: Video Games", "16": "Entertainment: Board Games", "17": "Science & Nature", "18": "Science: Computers",
                        "19": "Science: Mathematics", "20": "Mythology", "21": "Sports", "22": "Geography", "23": "History", "24": "Politics", "25": "Art", "26": "Celebrities", "27":
                        "Animals", "28": "Vehicles", "29": "Entertainment: Comics", "30": "Science: Gadgets", "31": "Entertainment: Japanese Anime & Manga", "32": "Entertainment: Cartoon &
                        Animations"}
  -n N

### CODE REVIEW

**Comment by dayeon3117** [comment]
Video link: https://duke.box.com/s/tmg3h4oxgskmjqxc76ixyyvyrfw2kdnc

**Comment by dayeon3117** [comment]
@medinardaniel Had a couple of questions:
1. Is there a specification on number of parameters to be added to the script?
2. Do we need to test this api for all edge cases, like in a software engineering assignment, or just the basic ones are enough?

**Comment by haranku16** [comment]
Instructions are clear and the script runs as expected to produce the CSV file!

File: api_trivia.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Resources used:
+# Requests API : https://realpython.com/python-requests/
+# Pandas API : https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html 
+#    (Along with related web-pages in pandas documentation for other references)
+# JSON API : https://www.geeksforgeeks.org/how-to-remove-key-value-pair-from-a-json-file-in-python/
+"""
```
Comment (haranku16) [inline api_trivia.py:6]: This section is super helpful so that the reader doesn't necessarily have to open the the PR to view usage instructions.

File: api_trivia.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    return pd.DataFrame(questions)
+
+
+def save_df_to_csv(dataframe, filename):
+    """
+    Saves the dataframe to csv format
+    """
+    dataframe.to_csv(filename, index = False)
+
+def parse_args(args):
```
Comment (haranku16) [inline api_trivia.py:102]: Would it be a good idea to add a docstring to this function too?

**Comment by brinnaebent** [comment]
Video 15/15. Good explanations, very clear demo.

**Review by mhy-666 — COMMENTED** [review]
Nice docstrings and clear functions!
